modules/layers/src/layers/editable-h3-cluster-layer.ts

```diff
@@ -0,0 +1,201 @@
+import EditableLayer from './editable-layer';
+
+const DEFAULT_EDIT_MODE = ViewMode;
+const DEFAULT_H3_RESOLUTION = 9;
+const EMPTY_FEATURE_COLLECTION = {
+  type: 'FeatureCollection',
+  features: [],
+};
+
+const defaultProps = {
+  mode: DEFAULT_EDIT_MODE,
+
+  // EditableGeoJsonLayer
+  ...EditableGeoJsonLayer.defaultProps,
+
+  // h3 layer
+  data: [],
+  selectedIndexes: [],
+  filled: false,
+  stroked: true,
+  lineWidthScale: 1,
+  lineWidthMinPixels: 1,
+  lineWidthMaxPixels: Number.MAX_SAFE_INTEGER,
+  lineWidthUnits: 'pixels',
+  getHexagons: (d) => d.hexIds,
+  getEditedCluster: (updatedHexagons, existingCluster) => {
+    if (existingCluster) {
+      return {
+        ...existingCluster,
+        hexIds: updatedHexagons,
+      };
+    }
+    return {
+      hexIds: updatedHexagons,
+    };
+  },
+  resolution: DEFAULT_H3_RESOLUTION,
+};
+
+export default class EditableH3ClusterLayer extends EditableLayer {
+  static layerName = 'EditableH3ClusterLayer';
+  static defaultProps = defaultProps;
+
+  initializeState() {
+    super.initializeState();
+
+    this.setState({
+      tentativeHexagonIDs: [],
+    });
+  }
+
+  // convert array of (lng, lat) coords to cluster of hexes
+  getDerivedHexagonIDs(coords) {
+    return polyfill(coords, this.props.resolution, true);
+  }
+
+  // convert pair of (lng, lat) coords into single hex
+  getDerivedHexagonID(coords) {
+    return geoToH3(coords[1], coords[0], this.props.resolution);
+  }
+
+  renderLayers() {
+    const layers: any = [
+      new EditableGeoJsonLayer(
+        this.getSubLayerProps({
+          id: 'editable-geojson',
+
+          mode: this.props.mode,
+          data: EMPTY_FEATURE_COLLECTION,
+          selectedFeatureIndexes: [],
+
+          onEdit: (editAction) => {
+            const { editType, editContext } = editAction;
+
+            switch (editType) {
+              case 'updateTentativeFeature':
+                // tentative feature updates, updated on every pointer move
+                if (editContext.feature.geometry.type === 'Polygon') {
+                  const coords = editContext.feature.geometry.coordinates;
+                  const hexIDs = this.getDerivedHexagonIDs(coords);
+
+                  this.setState({ tentativeHexagonIDs: hexIDs });
+                } else if (editContext.feature.geometry.type === 'Point') {
+                  const coords = editContext.feature.geometry.coordinates;
+                  const hexID = this.getDerivedHexagonID(coords);
+
+                  this.setState({ tentativeHexagonIDs: [hexID] });
+                }
+                break;
+              case 'addFeature':
+                // once the shape is finished drawing, calculate final hexagons
+
+                // get selected h3 cluster to act upon
+                // if there is more than 1 selected,
+                // throw an error, since it is ambiguous how to proceed
+                if (this.props.selectedIndexes.length > 1) {
```

Comment:
To make this more consistent with `EditableGeoJsonLayer`, can we update the logic to this?
```
if (!modeConfig || !modeConfig.booleanOperation) {
  // add new hex cluster
} else if (this.props.selectedIndexes.length !== 1) {
  // eslint-disable-next-line no-console,no-undef
  console.warn('booleanOperation only supported for single cluster selection');
  return;
} else {
  // switch on booleanOperation and affect single selected cluster
}
this.props.onEdit(...)
```